src/os/bluestore/bluefs_types.h

```diff
@@ -242,6 +279,21 @@ struct bluefs_transaction_t {
     encode((__u8)OP_FILE_UPDATE, op_bl);
     encode(file, op_bl);
   }
+  void op_file_update_inc(const bluefs_fnode_t& file) {
+    using ceph::encode;
+    encode((__u8)OP_FILE_UPDATE_INC, op_bl);
+    bluefs_fnode_delta_t delta;
+    delta.ino = file.ino;
+    delta.size = file.size;
+    delta.mtime = file.mtime;
+    uint64_t new_extents_size = 0;
+    for (const auto &p : file.new_extents) {
+      new_extents_size += p.length;
+    }
+    delta.offset = file.allocated - new_extents_size;
+    delta.extents = file.new_extents;
```

Comment:
nit: I'd like to avoid new_extents copying here and may be swap delta.extents and file.new_extents here or something. file.new_extents are cleared anyway afterwards. 
Something more elegant than doing that swap explicitly here is preferred though..
May be delta to be produced and filled by bluefs_fnode_t instance, e.g. via bluefs_fnode_t::make_delta or bluefs_fnode_t::branch_delta_off method?
This would eliminate the need to call clear_incremental_extents each time as well.